APE9.rst

```diff
@@ -0,0 +1,273 @@
+create the NDArr object with options not to propagate one or the other or 
+both to forestall complaints from those for which these operations will be 
+considered invalid.
+
+NDArr will support optional units since the propagation of units is
+unambiguous in mathematical operations.
+
+Supported numerical operations for NDData are: (+,-,*,/). While the discussion
+precluded use of ufuncs, there is no intrinsic reason ufuncs cannot be
+supported, so this APE will presume they are supported (eventually; there 
+is no implied commitment to support this functionality immediately). Ufuncs
+shall only work on dimensionless units if units are present.
+
+NDArr objects will support simple slicing (no striding, at least in the
+initial proposal). Slices will be based on array indices and no other
+interpretation.
+
+NDArr objects will retain information about their parent if derived through
+a slicing operation. This information shall include a pointer to the parent
+NDArr object, and the bounding box used to obtain the slice. When involving 
+binary arithmetic operations, between two NDArr objects that have different
+parents (or are themselves distinct parents), there will be no pointer to
+the original parent(s) present, nor will the bounding box information be
+retained
+
+If operations are unary, parentage and bounding box information is propagated.
+For binary operations with two NDArr objects, operations
+will retain parentage and bounding boxes only if both operands share the same
+parents and bounding box. If they don't, this information will be removed from
+the result.
+
+The NDArr object will permit indexing relative to the parent through a special
+attribute (details at 11).
+
+NDData Details
+++++++++++++++
+
+Composition instead of Inheritance
+----------------------------------
+
+NDData will not be a subclass of NDArr. Instead it will contain references
+to an NDArr object. To avoid confusion, the attribute that contains the 
+NDArr object will be .arr
+
+Need for ExposureInfo object?
+-----------------------------
+
+This is used by LSST to store information about the exposure such as filters,
+exposure time, etc. but all that is really needed is an isomorphic mapping
+to corresponding information in the astropy object. The question is: does
+it make sense to encapsulate that information in a single attribute, and 
+whether all think that these attributes are "astropy-universal”. For the 
+purposes of this, the only standard attribute is most likely the WCS.
+
+Items such as exposure time and filters could be relegated to the meta
+attribute. Making attributes such as these at the top level could be done
+through subclasses, though it may lead to many variants. Perhaps the best way
+to deal with this is for applications or libraries to state their requirements
+for items required to be in meta and leave it at that.
```

Comment:
I would have thought it will be useful to have more functionality than just a meta dictionary and WCS. While it's not at all practical to standardize everything comprehensively, some sort of fairly general meta-data abstraction (like Gemini has in our mostly-internal AstroData class) would help write data reduction routines that can be used with data from different instruments etc. (or at least avoid hard-wiring more assumptions than are necessary). I thought we also came up with some useful attributes to parse into objects besides WCS at the meeting but will have to dig out the photo. One thing I wonder about, off hand, is tracking associations between NDData objects besides parent-subarray (eg. calibrations that go with a science exposure). By the way, is the implied alternative to an ExposureInfo object here just to keep meta in NDData?